lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+
+Catbox has two interfaces.
+
+
+### Client
+[Client](https://github.com/hapijs/catbox#client) is low-level interface, which basically allows you set/get key-value pairs. Its initialized with one of the available adapters ([Memory](https://github.com/hapijs/catbox-memory), [Redis](https://github.com/hapijs/catbox-redis), [mongoDB](https://github.com/hapijs/catbox-mongodb), [Memcached](https://github.com/hapijs/catbox-memcached), [Riak](https://github.com/DanielBarnes/catbox-riak), [Amazon S3](https://github.com/fhemberger/catbox-s3)).
+
+hapi always initialize one default [client](https://github.com/hapijs/catbox#client) with [memory](https://github.com/hapijs/catbox-memory) adapter. Lets see how we can define more clients.
+
+```javascript
+var Hapi = require('hapi');
+
+var server = new Hapi.Server({
+    cache: [
+        {
+            name: 'mongoCache',
+            engine: require('catbox-mongodb'),
+            host: '127.0.0.1',
+            partition: 'cache',
+        },
+        {
+            name: 'redisCache',
+            engine: require('catbox-redis'),
+            host: '127.0.0.1',
+            partition: 'cache'
+       }
+    ]
+});
+
+server.connection({
+    port: 8000
+});
+```
+
+In this example two [clients](https://github.com/hapijs/catbox#client) were defined: `'mongoCache'`, `'redisCache'` and also default one without name with [memory](https://github.com/hapijs/catbox-memory) adapter. You can replace default client by omitting `name` property. `partition` property tells adapter how cache should be named (`'catbox'` by default). In case of [mongoDB](http://www.mongodb.org/) it is database name and in case of [redis](http://redis.io/) it is used as key prefix.
+
+### Policy
+[Policy](https://github.com/hapijs/catbox#policy) is high-level interface that you will be using. Lets pretend we are dealing with something more complicated than adding two numbers and we want to cache the results. [server.cache](http://hapijs.com/api#servercacheoptions) method creates new [policy](https://github.com/hapijs/catbox#policy), which is than used in handler. `id` object has mandatory key `id`, which is a unique item identifier string.
+
+```javascript
+var add = function (a, b, next) {
+
+    return next(null, Number(a) + Number(b));
+};
+
+var sumCache = server.cache({
+    cache: 'mongoCache',
+    expiresIn: 20 * 1000,
+    segment: 'customSegment',
+    generateFunc: function (id, next) {
+
+        add(id.a, id.b, next);
+    }
+});
+
+server.route({
+    path: '/add/{a}/{b}',
+    method: 'GET',
+    handler: function (request, reply) {
+
+        var id = request.params.a + ':' + request.params.b;
+        sumCache.get({id: id, a: request.params.a, b: request.params.b}, function (err, result) {
```

Comment:
Don't you only need the `id` key here in this object? I don't think `a` and `b` do anything?